Realtek IEEE 802.11 wireless network driver (rtwn) fails to start on purecap kernel

The driver has no issues starting on the default kernel:
Autoloading module: if_rtwn_usb
rtwn0 on uhub0
rtwn0: <Realtek 802.11n WLAN NIC, class 0/0, rev 2.00/0.00, addr 1> on usbus0

wlan0: flags=8843<UP,BROADCAST,RUNNING,SIMPLEX,MULTICAST> metric 0 mtu 1500
        ether 8c:ae:4c:f8:a5:9e
        inet 192.168.0.39 netmask 0xffffff00 broadcast 192.168.0.255
        groups: wlan

But it fails on purecap:
Autoloading module: if_rtwn_usb
rtwn0 on uhub0
rtwn0: <Realtek 802.11n WLAN NIC, class 0/0, rev 2.00/0.00, addr 1> on usbus0
  x0: 0xffff00019de07a30 [rR,0xffff00019de07a30-0xffff00019de07a3c] ($d.2 + b6)
  x1: 0xffff00019de07606 [rR,0xffff00019de07606-0xffff00019de07608] ($d.2 + 112)
  x2: 0xffff0000fe42ce16 [rwxRW,0xffff0000fe42cdf0-0xffff0000fe42ce30] (__bss_end + fce39a76)
  x3: 0xffff0000fe42ce17 [rwxRW,0xffff0000fe42cdf0-0xffff0000fe42ce30] (__bss_end + fce39a77)
  x4: 0x0000000000000041
  x5: 0xffff000000564159 [rxR,0x0000000000000000-0xffffffffffffffff] (kvprintf + 244)
  x6: 0xffff0000fe42cde0 [rRW,0xffff0000fe42cdc0-0xffff0000fe42cde0] (__bss_end + fce39a40)
  x7: 0xffff000000a63298 [rxR,0x0000000000000000-0xffffffffffffffff] (null_method + 48)
  x8: 0x0000000000008d60
  x9: 0x0000000000000018
 x10: 0x0000000000000000
 x11: 0x00000000fffffffe
 x12: 0x0000000000000002
 x13: 0x0000000000000000
 x14: 0x00000000ffffffd5
 x15: 0x0000000000000000
 x16: 0x0000000000000000
 x17: 0x0000000000000020
 x18: 0xffff0000fe42cdc0 [rwxRW,0xffff0000fe429000-0xffff0000fe42e000] (__bss_end + fce39a20)
 x19: 0xffffa0001c477000 [rwxRW,0xffffa0001c477000-0xffffa0001c477180]
 x20: 0xffff00019f25c000 [rwxRW,0xffff00019f25c000-0xffff00019f2651c0] (_DYNAMIC + 1352c60)
 x21: 0xffff0000fe42d150 [rwxRW,0xffff0000fe42d150-0xffff0000fe42d1c0] (__bss_end + fce39db0)
 x22: 0xffff00019f25c000 [rwxRW,0xffff00019f25c000-0xffff00019f26c000] (_DYNAMIC + 1352c60)
 x23: 0xffffa0001c477110 [rwxRW,0xffffa0001c477000-0xffffa0001c477180]
 x24: 0xffffa00051047200 [rwxRW,0xffffa00051047200-0xffffa00051047300]
 x25: 0xffff0000fe42cdf0 [rwxRW,0xffff0000fe42cdf0-0xffff0000fe42ce30] (__bss_end + fce39a50)
 x26: 0xffffa0000005d5f0 [rwxRW,0xffffa0000005d5f0-0xffffa0000005d600]
 x27: 0xffff00019f264d60 [rwxRW,0xffff00019f25c000-0xffff00019f2651c0] (_DYNAMIC + 135b9c0)
 x28: 0xffff000000a40125 [rR,0xffff000000a40125-0xffff000000a4013a] (aarch32_esigcode + dfefd)
 x29: 0xffff0000fe42ce30 [rwxRW,0xffff0000fe429000-0xffff0000fe42e000] (__bss_end + fce39a90)
 ddc: 0x0000000000000000
  sp: 0xffff0000fe42cdc0 [rwxRW,0xffff0000fe429000-0xffff0000fe42e000]
  lr: 0xffff00019de1d9d5 [rxR,0xffff00019de00000-0xffff00019de39000] (sentry) (rtwn_usb_attach + c4)
 elr: 0xffff00019de1d9f8 [rxR,0xffff00019de00000-0xffff00019de39000] (rtwn_usb_attach + e7)
spsr:         24400045
 far: ffff00019f2652b0
 esr:         9600006a
panic: Capability abort from kernel space: bounds violation
cpuid = 3
time = 1674743276
KDB: stack backtrace:
db_trace_self() at db_trace_self
db_trace_self_wrapper() at db_trace_self_wrapper+0x38
vpanic() at vpanic+0x1bc
panic() at panic+0x40
cap_abort() at cap_abort+0x250
handle_el1h_sync() at handle_el1h_sync+0x10
--- exception, esr 0x9600006a
rtwn_usb_attach() at rtwn_usb_attach+0xe8
device_attach() at device_attach+0x44c
device_probe_and_attach() at device_probe_and_attach+0x94
usb_probe_and_attach() at usb_probe_and_attach+0x318
uhub_explore() at uhub_explore+0x8bc
usb_bus_explore() at usb_bus_explore+0x11c
usb_process() at usb_process+0x134
fork_exit() at fork_exit+0x84
fork_trampoline() at fork_trampoline+0x10
KDB: enter: panic
[ thread pid 15 tid 100055 ]
Stopped at      kdb_enter+0x63: undefined       c200001f
db>

--- a/sys/dev/rtwn/pci/rtwn_pci_var.h
+++ b/sys/dev/rtwn/pci/rtwn_pci_var.h
@@ -102,7 +102,7 @@ enum {
 	 RTWN_PCI_INTR_RX_DESC_UNAVAIL | RTWN_PCI_INTR_RX_DONE)
 
 struct rtwn_pci_softc {
-	struct rtwn_softc	pc_sc;		/* must be the first */
+	struct rtwn_softc	pc_sc __subobject_member_used_for_c_inheritance;		/* must be the first */
 
 	struct resource		*irq;
 	struct resource		*mem;

--- a/sys/dev/rtwn/usb/rtwn_usb_var.h
+++ b/sys/dev/rtwn/usb/rtwn_usb_var.h
@@ -57,7 +57,7 @@ enum {
 #define RTWN_EP_QUEUES		RTWN_BULK_RX
 
 struct rtwn_usb_softc {
-	struct rtwn_softc	uc_sc;		/* must be the first */
+	struct rtwn_softc	uc_sc __subobject_member_used_for_c_inheritance;		/* must be the first */
 	struct usb_device	*uc_udev;
 	struct usb_xfer		*uc_xfer[RTWN_N_TRANSFER];
 

